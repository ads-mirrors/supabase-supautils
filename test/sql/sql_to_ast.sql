select supa.sql_to_ast($$
    select name, id, dob
    from public.uSer
    where id = 'hello\nworld'
    limit 1
$$::text);


select supa.sql_to_ast($$
    create table acct(
        id int primary key
    )
$$::text);
