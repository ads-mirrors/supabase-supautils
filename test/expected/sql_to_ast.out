select supa.sql_to_ast($$
    select name, id, dob
    from public.uSer
    where id = 'hello\nworld'
    limit 1
$$::text);
            sql_to_ast             
-----------------------------------
 (                                +
    {RAWSTMT                      +
    :stmt                         +
       {SELECT                    +
       :distinctClause <>         +
       :intoClause <>             +
       :targetList (              +
          {RESTARGET              +
          :name <>                +
          :indirection <>         +
          :val                    +
             {COLUMNREF           +
             :fields ("name")     +
             :location 12         +
             }                    +
          :location 12            +
          }                       +
          {RESTARGET              +
          :name <>                +
          :indirection <>         +
          :val                    +
             {COLUMNREF           +
             :fields ("id")       +
             :location 18         +
             }                    +
          :location 18            +
          }                       +
          {RESTARGET              +
          :name <>                +
          :indirection <>         +
          :val                    +
             {COLUMNREF           +
             :fields ("dob")      +
             :location 22         +
             }                    +
          :location 22            +
          }                       +
       )                          +
       :fromClause (              +
          {RANGEVAR               +
          :schemaname public      +
          :relname user           +
          :inh true               +
          :relpersistence p       +
          :alias <>               +
          :location 35            +
          }                       +
       )                          +
       :whereClause               +
          {AEXPR                  +
          :name ("=")             +
          :lexpr                  +
             {COLUMNREF           +
             :fields ("id")       +
             :location 57         +
             }                    +
          :rexpr                  +
             {A_CONST             +
             :val "hello\\nworld" +
             :location 62         +
             }                    +
          :location 60            +
          }                       +
       :groupClause <>            +
       :havingClause <>           +
       :windowClause <>           +
       :valuesLists <>            +
       :sortClause <>             +
       :limitOffset <>            +
       :limitCount                +
          {A_CONST                +
          :val 1                  +
          :location 87            +
          }                       +
       :lockingClause <>          +
       :withClause <>             +
       :op 0                      +
       :all false                 +
       :larg <>                   +
       :rarg <>                   +
       }                          +
    :stmt_location 0              +
    :stmt_len 0                   +
    }                             +
 )                                +
 
(1 row)

select supa.sql_to_ast($$
    create table acct(
        id int primary key
    )
$$::text);
                sql_to_ast                
------------------------------------------
 (                                       +
    {RAWSTMT                             +
    :stmt                                +
       {CREATESTMT                       +
       :relation                         +
          {RANGEVAR                      +
          :schemaname <>                 +
          :relname acct                  +
          :inh true                      +
          :relpersistence p              +
          :alias <>                      +
          :location 18                   +
          }                              +
       :tableElts (                      +
          {COLUMNDEF                     +
          :colname id                    +
          :typeName                      +
             {TYPENAME                   +
             :names ("pg_catalog" "int4")+
             :typeOid 0                  +
             :setof false                +
             :pct_type false             +
             :typmods <>                 +
             :typemod -1                 +
             :arrayBounds <>             +
             :location 35                +
             }                           +
          :inhcount 0                    +
          :is_local true                 +
          :is_not_null false             +
          :is_from_type false            +
          :storage <>                    +
          :raw_default <>                +
          :cooked_default <>             +
          :identity <>                   +
          :identitySequence <>           +
          :generated <>                  +
          :collClause <>                 +
          :collOid 0                     +
          :constraints (                 +
             {CONSTRAINT                 +
             :conname <>                 +
             :deferrable false           +
             :initdeferred false         +
             :location 39                +
             :contype PRIMARY_KEY        +
             :keys <>                    +
             :including <>               +
             :options <>                 +
             :indexname <>               +
             :indexspace <>              +
             :reset_default_tblspc false +
             }                           +
          )                              +
          :fdwoptions <>                 +
          :location 32                   +
          }                              +
       )                                 +
       :inhRelations <>                  +
       :partspec <>                      +
       :partbound <>                     +
       :ofTypename <>                    +
       :constraints <>                   +
       :options <>                       +
       :oncommit 0                       +
       :tablespacename <>                +
       :accessMethod <>                  +
       :if_not_exists false              +
       }                                 +
    :stmt_location 0                     +
    :stmt_len 0                          +
    }                                    +
 )                                       +
 
(1 row)

